### PR TITLE
Fix dev-docs about authorization, again

### PR DESCRIPTION
**Pull request by lbam**
The dev-docs incorrectly describe the scope checking as performing an AND of scopes at different levels. That turns out to be false. Only the highest-level scopes are checked.

Also, the dev-docs were kind of fuzzy about the exact workings of profiles. This explains them in some more detail wrt. requests, scopes and schemas. The "statistician" example now explains exactly how schemas fall short (lack of an OR operator, really).

In the example, I've replaced "encoded" by "read" because "encoded" isn't actually implemented in dso-api.

### CODE REVIEW

File: dev-docs/source/auth.rst [hunk 1]
```diff
@@ -23,13 +23,14 @@ At every level, the ``auth`` field contains a string that is called a *scope*.
...
 to meet an ``auth`` field's requirements.
-The absence of an ``auth`` field marks a publicly available resource.
+The absence of a scope marks a publicly available resource.
 
-The presence of ``auth`` fields on multiple levels
-means that all listed scopes must be carried in the JWT.
-So, if the dataset has ``"auth": "FOO/BAR"`` and a field has ``"auth": "FOO/BAZ``,
-then access to the field is only available
-with a token that carries the ``FOO/BAR`` and ``FOO/BAZ`` scopes.
+Scopes at higher levels override scopes at lower levels.
```
Comment (ngaranko) [inline dev-docs/source/auth.rst:28]: *higher levels of schema.
Otherwise it may be confusing people that scopes also have hierarchy, which they do not really.

File: dev-docs/source/auth.rst [hunk 2]
```diff
@@ -23,13 +23,14 @@ At every level, the ``auth`` field contains a string that is called a *scope*.
 A request's JWT must contain this scope in its ``scopes`` claim
 (see the ``authorization_django`` source code for details)
 to meet an ``auth`` field's requirements.
-The absence of an ``auth`` field marks a publicly available resource.
+The absence of a scope marks a publicly available resource.
```
Comment (ngaranko) [inline dev-docs/source/auth.rst:26]: This was correct, isn't it?

Comment (lbam) [reply dev-docs/source/auth.rst:26]: I thought it wasn't clear enough, but my edit actually made it less clear. Let me try again.

File: dev-docs/source/auth.rst (hunk 1 above)
Comment (lbam) [reply dev-docs/source/auth.rst:28]: Good point.

File: dev-docs/source/auth.rst (hunk 2 above)
Comment (lbam) [reply dev-docs/source/auth.rst]: Restored to the old text.

File: dev-docs/source/auth.rst [hunk 3]
```diff
@@ -71,18 +78,28 @@ The file format provides various options, such as:
...
 may only access data of someone when they can provide their last name and postal code.
-
-Another case for profiles would be a statistician, that may only read the age and neighbourhood
+Or a statistician might be allowed to read age and neighbourhood
 fields to aggregate data, without ever having access to identifiable data.
-Encoding such feature into the schema file amongst other rules would produce
-very complex ``auth`` fields, while the profile file gives a
-fitting definition of such role and permissions.
+Encoding such rules in the schema file using a custom scope
+would require every other request that accesses the dataset to also have this scope.
```
Comment (jjmurre) [inline dev-docs/source/auth.rst:103]: May add some explanation about what `"permission": "read"` exactly means compared to `"field": {"bsn": "read"}`.

Comment (lbam) [reply dev-docs/source/auth.rst:103]: I honestly don't know what it means.

File: dev-docs/source/auth.rst [hunk 4]
```diff
@@ -25,11 +25,12 @@ A request's JWT must contain this scope in its ``scopes`` claim
...
-then access to the field is only available
-with a token that carries the ``FOO/BAR`` and ``FOO/BAZ`` scopes.
+Scopes at higher levels of the schema override scopes at lower levels.
+That is, a dataset scope overrides a table scope
+and a table scope overrides field scopes.
+So, if the dataset has ``"auth": "FOO/BAR"``,
+then this scope is required for any access to the dataset
+and any lower-level scopes are ignored.
 
 The effect of an ``auth`` on a dataset or table is that attempts to access it
```
Comment (jjmurre) [inline dev-docs/source/auth.rst:35]: Maybe add: To access it "without the proper scope"  results in ...

File: dev-docs/source/auth.rst (hunk 3 above)
Comment (ngaranko) [reply dev-docs/source/auth.rst:103]: Permission "read" means user can see original value of the field in database. (e.g. default for all). It is used to give access to individual fields using Profile.

**Review by jjmurre — APPROVED** [review]
Clear explanation!  One other small thing: line 9 of auth.rst mentions "These scopes", but that does not refer to scopes mentioned before in the text.

**Review by ngaranko — APPROVED** [review]

**Comment by lbam** [comment]
Thanks!
